server/planning/events/events.py

```diff
@@ -128,6 +138,20 @@ def is_event_updated(new_item: Event, old_item: Event) -> bool:
     return False
 
 
+def get_user_updated_keys(id: str) -> set[str]:
+    # Placeholder function definition
+    # Replace this with the actual implementation
+    history_service = get_resource_service("events_history")
+    updates = history_service.get_by_id(id)
+    updated_keys = set()
+    for update in updates:
+        if not update.get("user_id"):
+            continue
+        for key in update.get("update", {}):
```

Comment:
Instead of using an iterator, you could use `updated_keys.update(update.get("update", {}).keys())`